HW12_iterators/iterators.py

```diff
@@ -0,0 +1,150 @@
+        self.n_seq = len(self.seq_list)
+        self.n_to_mutate = n_to_mutate
+        self.cur_index = -1
+
+        self.iteration_list = [[x, y] for x, y in zip(self.ids_list, self.seq_list)]
+
+        if self.modification == 'deletion':
+            self.deletion()
+
+        elif self.modification == 'random mutation':
+            self.random_mutation()
+
+        if set(self.seq_list[0]).issubset(self.only_aa_set):
+            self.type = 'protein'
+            self.list_for_choice = self.aa_list
+        else:
+            self.type = 'nucleic_acid'
+            self.list_for_choice = self.nucleotide_list
+
+    def decision(self):
+        return random.random() < self.probability
+
+    def deletion(self):
+
+        seq_list_to_print = self.seq_list
+
+        for i in range(self.n_seq):
+            if self.decision():  # probability check
+                deletion_start = random.randint(0, len(self.seq_list[i]) - self.n_to_mutate)
+                deletion_end = deletion_start + self.n_to_mutate
+                seq_list_to_print[i] = self.seq_list[i][:deletion_start] + self.seq_list[i][deletion_end:]
+
+        self.iteration_list = [[x, y] for x, y in zip(self.ids_list, seq_list_to_print)]
+
+    def random_mutation(self):
+
+        seq_list_to_print = self.seq_list
+
+        for i in range(self.n_seq):
+            if self.decision():  # probability check
+
+                for i in range(self.n_to_mutate):  # we need n_aa mutations
+                    mutation_site = random.randint(0, len(self.seq_list[i]) - 1)  # choose mutation site
+                    seq_list_to_print[i] = (self.seq_list[i][:mutation_site]
+                                            + random.choice(self.list_for_choice)
+                                            + self.seq_list[i][mutation_site + 1:])
+
+        self.iteration_list = [[x, y] for x, y in zip(self.ids_list, seq_list_to_print)]
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+
+        self.cur_index += 1
+
+        if self.cur_index == self.n_seq:
+            self.cur_index = 0
+            if self.modification == 'deletion':
+                self.deletion()
+
+            elif self.modification == 'random mutation':
+                self.random_mutation()
+
+        return self.iteration_list[self.cur_index][0] + " " + self.iteration_list[self.cur_index][1]
+
+
+# 1*. Create a generator iter_append(iterable, item), that "add" element item in the "end" of iterable.
+
+
+def iter_append(iterable, item):
+    yield from iterable
+    yield item
+
+
+# 2*. Create a function, that "unzip" nested lists.
+
+
+def generator_from_nested_lists(list1):
+    for i in list1:
+        if isinstance(i, list):
+            yield from generator_from_nested_lists(i)
+        else:
+            yield i
+
+
+def nested_lists_unpacker(list1):
+    list_to_print = []
+    for i in generator_from_nested_lists(list1):
+        list_to_print.append(i)
+
+    return list_to_print
```

Comment:
Альтернативный вариант
```suggestion
    return list(generator_from_nested_lists(list1))
```